modules/xmpp/moderator.js

```diff
@@ -90,100 +109,64 @@ export default function Moderator(roomName, xmpp, emitter, options) {
-};
-
-Moderator.prototype.onMucMemberLeft = function(jid) {
-    const resource = Strophe.getResourceFromJid(jid);
-
-    if (resource === 'focus') {
-        logger.info(
-            'Focus has left the room - leaving conference');
-        this.eventEmitter.emit(XMPPEvents.FOCUS_LEFT);
+Moderator.prototype.isFocusJid = function(jid) {
+    if (!jid) {
+        return false;
     }
-};
 
-Moderator.prototype.setFocusUserJid = function(focusJid) {
-    if (!this.focusUserJid) {
-        this.focusUserJid = focusJid;
-        logger.info(`Focus jid set to:  ${this.focusUserJid}`);
+    for (const focusJid of this.focusUserJids) {
+        // jid may be a full JID, and focusUserJids may be bare JIDs
+        if (jid.indexOf(`${focusJid}/`) === 0) {
+            return true;
+        }
     }
-};
 
-Moderator.prototype.getFocusUserJid = function() {
-    return this.focusUserJid;
+    return false;
 };
 
-Moderator.prototype.getFocusComponent = function() {
-    // Get focus component address
-    let focusComponent = this.options.connection.hosts.focus;
-
-    // If not specified use default:  'focus.domain'
-
-    if (!focusComponent) {
-        focusComponent = `focus.${this.options.connection.hosts.domain}`;
-    }
+Moderator.prototype.isExternalAuthEnabled = function() {
+    return this.externalAuthEnabled;
+};
 
-    return focusComponent;
+Moderator.prototype.isSipGatewayEnabled = function() {
+    return this.sipGatewayEnabled;
 };
 
-Moderator.prototype.createConferenceIq = function() {
-    // Generate create conference IQ
-    const elem = $iq({ to: this.getFocusComponent(),
-        type: 'set' });
+/**
+ * Create a conference request based on the configured options and saved Settings.
+ *
+ * A conference request has the following format:
+ * {
+ *   room: "room@example.com",
+ *   sessionId: "foo", // optional
+ *   machineUdi: "bar", // optional
+ *   identity: "baz", // optional
+ *   properties: { } // map string to string
+ * }
+ *
+ * It can be encoded in either JSON or and IQ.
+ *
+ * @returns the created conference request.
+ */
+Moderator.prototype._createConferenceRequest = function() {
 
     // Session Id used for authentication
-    const { sessionId } = Settings;
-    const machineUID = Settings.machineId;
-    const config = this.options.conference;
-
-    logger.info(`Session ID: ${sessionId} machine UID: ${machineUID}`);
-
-    elem.c('conference', {
-        xmlns: 'http://jitsi.org/protocol/focus',
-        room: this.roomName,
-        'machine-uid': machineUID
-    });
-
-    if (sessionId) {
-        elem.attrs({ 'session-id': sessionId });
-    }
+    const { sessionId, machineUID } = Settings;
+    const config = this.options;
+    const properties = {};
 
     if (config.startBitrate) {
-        elem.c(
-            'property', {
-                name: 'startBitrate',
-                value: config.startBitrate
-            }).up();
+        properties.startBitrate = config.startBitrate;
```

Comment:
This is passed to jicofo, which then uses it when generating offers for everyone in the conference. It's an awkward flow, especially since every participant sends it, but only the first one takes effect, but it's how we currently allow this to be configured per-conference. I would be very happy if we can make it a local client option, but I don't think we can just drop it.

Reply:
oh I see. Looks like "x-google-start-bitrate=800" is set in the payload info. Do we know if this has any effect on the target encode bitrate set on the encoder ? We have maxBitrates set on the sender encodings which I believe overrides this value. Can we drop it in Jicofo and then the client can stop sending it ?